oic.wk.rd.raml

```diff
@@ -0,0 +1,166 @@
+#%RAML 0.8
+#RAML for Resource Directory
+title: Resource Directory
+version: v1-20160622
+documentation:
+
+  - title: OCF Copyright
+    content: Copyright (c) 2016, 2017 Open Connectivity Foundation, Inc. All rights reserved.
+  - title: OCF License
+    content: !include LICENSE.md
+  - title: OCF Disclaimer
+    content: !include DISCLAIMER.md
+
+schemas:
+  # rdSelection represents the payload of GET response.  
+  # rdPublish represents the payload in POST request. 
+-  rdSelection: !include schemas/oic.rd.selection-schema.json
+-  rdPublish: !include schemas/oic.rd.publish-schema.json
+	
+/oic/rd:
+  displayName: Resource directory resource
+  description: |
+    Resource to be exposed by any Device that can act as a Resource Directory.
+    1) Provides selector criteria (e.g. integer) with GET request 
+    2) Publish or Update a Link in /oic/res with POST request 
+    3) Delete a Link in /oic/res with DELETE request 
+
+  get:
+    description: |
+      Get the attributes of the Resource Directory for selection purposes.
+    queryParameters:
+      if:
+        description: "The Interface to be used."
+        enum: ["oic.if.baseline"]
+    responses:
+      200:
+        description: |
+          Respond with the selector criteria - either the set of attributes or the bias factor
+        body:
+          application/json:
+            schema: rdSelection
+            example: |
+              {
+                "rt": ["oic.wk.rd"],
+		"if": ["oic.if.baseline"], 
+                "sel": 50
+              }
+
+  post:
+    description: |
+      Publish the resource information for the first time or Update the existing one in /oic/res. 
+      Appropriates parts of the information, i.e. Links of the published Resources will be discovered through /oic/res. 
+      1) When a Device first publishes a Link, the request payload to RD may include the Links without "ins" Parameter.  	 
```

Comment:
Upon CREATION, a publishing Device may include "ins" with its preferred value or 
omit "ins" altogether so that the receiving RD can assign a unique "ins" value.